App.js

```diff
@@ -51,42 +74,41 @@ export default class App extends Component {
       }
     );
 
-    const sdkInitialized = await this.isSdkInitialized();
-    if (sdkInitialized) {
-      this.sdkCrashEventSubscription = rnSentianceEmitter.addListener(
-        "SDKCrashEvent",
-        ({ time, lastKnownLocation }) => {
-          this.setState({
-            crash: {
-              time: new Date(time),
-              lastKnownLocation,
-            },
-          });
-        }
-      );
-      RNSentiance.listenCrashEvents();
-    }
+    this.sdkCrashEventSubscription = rnSentianceEmitter.addListener(
+      "SDKCrashEvent",
+      ({ time, lastKnownLocation }) => {
+        this.setState({ crash: { time: new Date(time), lastKnownLocation } });
+      }
+    );
   }
 
   async setupSdk() {
     const sdkInitialized = await this.isSdkInitialized();
-    if (sdkInitialized) {
-      const sdkStatus = await RNSentiance.getSdkStatus();
-      const userId = await RNSentiance.getUserId();
-      const sdkVersion = await RNSentiance.getVersion();
-      const data = await this.statusToData(sdkStatus);
-      this.setState({ userId, sdkVersion, data });
-      RNSentiance.listenUserActivityUpdates();
-    } else {
-      await this.initializeSDK();
+    if (!sdkInitialized) {
+      const appId = "{{APP_ID}}";
+      const appSecret = "{{APP_SECRET}}";
+
+      await RNSentiance.initWithUserLinkingEnabled(
+        appId,
+        appSecret,
+        null,
+        true
+      );
     }
+
+    const userId = await RNSentiance.getUserId();
```

Comment:
If init fails above, the `getUserId()` call seem to trigger `promise.reject()` in the RN plugin. What is assigned to `userId` then?

Reply:
If `initWithUserLinkingEnabled` failed above, the error will be thrown immediately. The next tick of the process will be this line https://github.com/sentiance/react-native-sentiance-example/pull/6/files/480ffa3486ed228ce912b2ee3b58535740dfa74d#diff-958f6f56cd42c904603d226cd36a57b6R32

So RNSentiance.getUserId(); won't be invoked.